pandas/core/internals.py

```diff
@@ -4517,6 +4551,45 @@ def fast_xs(self, loc):
         """
         return self._block.values[loc]
 
+    def concat(self, to_concat, new_axis):
+        """
+        Concatenate a list of SingleBlockManagers into a single
+        SingleBlockManager.
+
+        Used for pd.concat of Series objects with axis=0.
+
+        Parameters
+        ----------
+        to_concat : list of SingleBlockManagers
+        new_axis : Index of the result
+
+        Returns
+        -------
+        SingleBlockManager
+
+        """
+        non_empties = [x for x in to_concat if len(x) > 0]
+
+        # check if all series are of the same block type:
+        if len(non_empties) > 0:
+            blocks = [obj.blocks[0] for obj in non_empties]
+
+            if all([type(b) is type(blocks[0]) for b in blocks[1:]]):  # noqa
```

Comment:
could some of this logic be moreve to concaty_same_type?

Reply:
Which logic do you mean exactly? 
For now, I coded `concat_same_type` such that is assumes only blocks of the same type are passed (so I don't do the checking there, but before the method is called)